openedxscorm/scormxblock.py

```diff
@@ -384,7 +384,7 @@ def extract_package(self, package_file):
                     # the is_dir() method to verify whether a ZipInfo object points to a
                     # directory.
                     # https://docs.python.org/3.6/library/zipfile.html#zipfile.ZipInfo.is_dir
-                    if not zipinfo.filename.endswith("/"):
+                    if not (zipinfo.filename.endswith("/") or zipinfo.filename.endswith("\\")):
```

Comment:
what if you use is_dir method to check if file is directory or not as mentioned in the inline comment since python 3.6+ is supported now? 
```
# Do not unzip folders, only files. In Python 3.6 we will have access to
                    # the is_dir() method to verify whether a ZipInfo object points to a
                    # directory.
                    # https://docs.python.org/3.6/library/zipfile.html#zipfile.ZipInfo.is_dir
```

Reply:
@ziafazal, the `is_dir` method checks for `/` at the end of the filename which is the same as condition `zipinfo.filename.endswith("/")` currently being used.

'is_dir' method implementation:
```
def is_dir(self):
    """Return True if this archive member is a directory."""
    return self.filename[-1] == '/'
```

From [python docs:](https://docs.python.org/3.8/library/zipfile.html#zipfile.ZipInfo.is_dir)
```
ZipInfo.is_dir()
    Return True if this archive member is a directory.

    This uses the entry’s name: directories should always end with /.

    New in version 3.6.
```

And, I found this filename `filename: 'plugins\\plugins\\'` in some of my scorm zip files.

Reply:
@talhaaslam01 it appears the issue lies with archives created using Windows OS having older version of PowerShell. That underlying [issue](https://github.com/PowerShell/Microsoft.PowerShell.Archive/issues/48) appears to be fixed in PowerShell and there is new `is_dir` method which addresses the [issue](https://github.com/python/cpython/issues/117084) in python. I'd suggest backporting `is_dir` method in our codebase until we upgrade to python 3.12.3 or greater which has a fix for windows based archives. Please add an inline TODO comment about removing the method once we upgrade to python 3.12.3 or greater.

```suggestion
 def is_dir(self):
        """Return True if this archive member is a directory."""
        if self.filename.endswith('/'):
            return True
        # The ZIP format specification requires to use forward slashes
        # as the directory separator, but in practice some ZIP files
        # created on Windows can use backward slashes.  For compatibility
        # with the extraction code which already handles this:
        if os.path.altsep:
            return self.filename.endswith((os.path.sep, os.path.altsep))
        return False
```

Reply:
@ziafazal, I have updated the code.